### PR TITLE
adding pdf conversion services and controllers

### PR DESCRIPTION
**Pull request by ocarleton**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FR-1855

### Change description ###
adding endpoint and service to take in a word document and return a PDF document

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/controller/DocumentConversionController.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            + "cannot be reached", response = String.class),
+        @ApiResponse(code = 500, message = "Returned when there is an unknown server error",
+            response = String.class)
+    })
+    @PostMapping("/version/1/convert-to-pdf")
+    public Document convertDocumentToPdf(
+        @RequestHeader(value = "Authorization") String authorisationToken,
+        @RequestBody Document document
+    ) {
+        log.info("document conversion service is null: {}", documentConversionService == null);
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/controller/DocumentConversionController.java:41]: testing log

File: src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/error/DocumentConversionException.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.hmcts.reform.finrem.documentgenerator.error;
+
+public class DocumentConversionException extends RuntimeException {
+    public DocumentConversionException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/error/DocumentConversionException.java:7]:
```suggestion
```

File: src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/service/DocumentConversionService.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
+package uk.gov.hmcts.reform.finrem.documentgenerator.service;
+
+
+
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/service/DocumentConversionService.java:4]:
```suggestion
```

**Review by shaunganley — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/controller/DocumentConversionController.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+@RestController
+@RequiredArgsConstructor
+@Api(value = "Document conversion", tags = {"Document conversion"})
+@Slf4j
+public class DocumentConversionController {
+
+    private final DocumentConversionService documentConversionService;
+    private final DocumentManagementService documentManagementService;
+
+    @ApiOperation(value = "Deletes document from the evidence management.", tags = {"Document Generation"})
```
Comment (jakub-kozlowski) [inline src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/controller/DocumentConversionController.java:26]: Is it actually deleting document from Evidence Management? Looks like it's storing a PDF version

File: src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/service/DocumentConversionService.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    @Value("${service.pdf-service.accessKey}")
+    private String docmosisAccessKey;
+
+    private final RestTemplate restTemplate;
+
+    private final EvidenceManagementService evidenceManagementService;
+
+
+    public byte[] convertDocumentToPdf(Document sourceDocument) {
+        if (sourceDocument.getFileName().toLowerCase().endsWith(PDF)) {
```
Comment (jakub-kozlowski) [inline src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/service/DocumentConversionService.java:44]: I'm not convinced checking document extension is a valid way of checking the file type. We already have Apache Tika in DocGenService that can be leveraged for that purpose: https://github.com/hmcts/finrem-document-generator-client/blob/master/src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/config/DocumentGeneratorApplicationConfiguration.java#L12

File: src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/service/DocumentConversionService.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                "Document already is a pdf",
+                null
+            );
+        }
+
+        return convert(sourceDocument, PDF);
+    }
+
+    public String getConvertedFilename(String filename) {
+        return filename.split("\\.")[0] + ".pdf";
```
Comment (jakub-kozlowski) [inline src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/service/DocumentConversionService.java:55]: Better use something from `org.apache.commons.io.FilenameUtils` - the service has the library already included. This version would return eg. "order.pdf" for "order.contested.doc"

**Review by jakub-kozlowski — CHANGES_REQUESTED** [review]

**Review by jakub-kozlowski — APPROVED** [review]

**Pull request merged**